pkg/components/ebpf/common/pids.go

```diff
@@ -215,16 +218,37 @@ func (pf *IdentityPidsFilter) Filter(inputSpans []request.Span) []request.Span {
 	return inputSpans
 }
 
-func checkIfExportsOTel(svc *svc.Attrs, span *request.Span) {
+func (pf *PIDsFilter) checkIfExportsOTel(svc *svc.Attrs, span *request.Span) {
 	if span.IsExportMetricsSpan() {
 		svc.SetExportsOTelMetrics()
+		pf.reportAvoidedService(svc, "metrics")
 	} else if span.IsExportTracesSpan() {
 		svc.SetExportsOTelTraces()
+		pf.reportAvoidedService(svc, "traces")
 	}
 }
 
-func checkIfExportsOTelSpanMetrics(svc *svc.Attrs, span *request.Span) {
+func (pf *PIDsFilter) checkIfExportsOTelSpanMetrics(svc *svc.Attrs, span *request.Span) {
 	if span.IsExportTracesSpan() {
 		svc.SetExportsOTelMetricsSpan()
+		pf.reportAvoidedService(svc, "metrics_span")
+	}
+}
+
+// reportAvoidedService calls the appropriate internal metrics method based on telemetry type
+func (pf *PIDsFilter) reportAvoidedService(svc *svc.Attrs, telemetryType string) {
```

Comment:
Do we need to check somewhere here that internal metrics are enabled so that we don't call this function if they are not on?

The reason this is very important is that this code will now run for every span that comes through. 

It also raises the question, do we want to count every time a span is processed? Seems very excessive for something that's only per service. I'd much rather have this check done when we detect that we have otel enabled service and then never again.